src/coreclr/jit/importer.cpp

```diff
@@ -8645,9 +8645,6 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
                    opcodeNames[opcode], callInfo->kind, varTypeName(callRetTyp), structSize);
         }
 #endif
-        // This should be checked in impImportBlockCode.
-        assert(!compIsForInlining() || !(impInlineInfo->inlineCandidateInfo->dwRestrictions & INLINE_RESPECT_BOUNDARY));
```

Comment:
I think you should be also able delete `dwRestrictions` field in `inlineCandidateInfo` and on JIT/EE interface.